tests/components/google/test_init.py

```diff
@@ -604,3 +608,48 @@ async def test_expired_token_requires_reauth(
     flows = hass.config_entries.flow.async_progress()
     assert len(flows) == 1
     assert flows[0]["step_id"] == "reauth_confirm"
+
+
+@pytest.mark.parametrize(
+    "calendars_config",
+    [
+        [
+            {
+                "cal_id": "ignored",
+                "entities": {"device_id": "existing", "name": "existing"},
+            }
+        ],
+        [],
+    ],
+    ids=["has_yaml", "no_yaml"],
+)
+async def test_calendar_yaml_update(
+    hass: HomeAssistant,
+    component_setup: ComponentSetup,
+    mock_calendars_yaml: Mock,
+    mock_calendars_list: ApiResult,
+    test_api_calendar: dict[str, Any],
+    mock_events_list: ApiResult,
+    setup_config_entry: MockConfigEntry,
+    calendars_config: dict[str, Any],
+) -> None:
+    """Test updating the yaml file with a new calendar."""
+
+    mock_calendars_list({"items": [test_api_calendar]})
+    mock_events_list({})
+    assert await component_setup()
+
+    mock_calendars_yaml().read.assert_called()
+    if calendars_config:
+        mock_calendars_yaml().write.assert_called()
```

Comment:
Can we parametrize the call count instead so we don't need conditions in the test?

Reply:
The # of calls is like 44 since the mock open gets invoked a bunch of times (e.g. one call for every newline in the file) so instead of parameterizing call count i've parameterized the boolean rather than relying on the other parameter so it's at least a little more explicit.   If there is a simple way to parameterize "zero" vs "not zero" let me know.